topics/climate/tutorials/pangeo-notebook/tutorial.md

```diff
@@ -0,0 +1,723 @@
+- 
+  type: "external"
+  title: Data Carpentry Geospatial Workshop
+  link: https://datacarpentry.org/geospatial-workshop/
+follow_up_training:
+- 
+  type: "external"
+  title: Pangeo Tutorial Gallery
+  link: http://gallery.pangeo.io/repos/pangeo-data/pangeo-tutorial-gallery/index.html
+- 
+  type: "external"
+  title: Pangeo Gallery
+  link: http://gallery.pangeo.io/index.html
+questions:
+- What is Pangeo notebook?
+- How to start Pangeo Notebook in Galaxy?
+- What are the main software components of the Pangeo ecosystem?
+- What is Xarray?
+- How to manipulate Xarrays?
+- How to print metadata information?
+- How to make a selection?
+- How to visualize?
+- How to filter? 
+- How to make reduction operations (mean, max, min)?
+- How to resample my data?
+- Where to go next?
+objectives:
+- Learn to get metadata information using Xarray Galaxy Tools
+- Learn to select data
+- Learn to visualize data
+- Learn to filter, make reduction operations (mean, max, min)
+- Learn to resample my data
+- Learn to cite and contribute to Pangeo
+time_estimation: 1H
+key_points:
+- Pangeo and Xarray
+- Read, select, mask and plot netCDF data using Xarray
+tags:
+  - pangeo
+  - interactive-tools
+  - jupyter-notebook
+contributors:
+- annefou
+notebook:
+  language: python
+  snippet: topics/climate/tutorials/pangeo-notebook/preamble.md
+---
+
+
+# Introduction
+{:.no_toc}
+
+In this tutorial, we will learn about [Xarray](https://xarray.pydata.org/), one of the most used Python library from the [Pangeo](https://pangeo.io/) ecosystem. 
+
+We will be using data from [Copernicus Atmosphere Monitoring Service](https://ads.atmosphere.copernicus.eu/)
+and more precisely PM2.5 ([Particle Matter < 2.5 μm](https://en.wikipedia.org/wiki/Particulates#Size,_shape_and_solubility_matter)) 4 days forecast from December, 22 2021. Parallel data analysis with Pangeo is not covered in this tutorial. 
+
+> ### {% icon comment %} Remark
+>
+> This tutorial uses data on a regular latitude-longitude grid. More complex and irregular grids are not discussed in this tutorial. In addition,
+> this tutorial is not meant to cover all the different possibilities offered by Xarrays but shows functionalities we find useful for day to day
+> analysis.
+>
+{: .comment}
+
+> ### Agenda
+>
+> In this tutorial, we will cover:
+>
+> 1. TOC
+> {:toc}
+>
+{: .agenda}
+
+
+# Analysis
+
+## Import Python packages
+
+Some packages may need to be installed first. For example `cmcrameri` is missing, so we need to do:
+
+```python
+pip install cmcrameri
+```
+then import all the necessary packages in your Jupyter Notebook.
+
+```python
+import numpy as np
+import xarray as xr
+import cartopy.crs as ccrs
+import matplotlib.pyplot as plt
+import cmcrameri.cm as cmc
+import pandas as pd
+```
+In case some packages cannot be imported you need to install them first. For example if cmcrameri is missing, open a terminal and type pip install cmcrameri, then re-execute the cell in your Jupyter Notebook.
```

Comment:
```suggestion
```

Reply:
Ok. I am the one you talked about terminal. Defintiely a mistake! I need to update this part. Everything needs to be installed in the notebook so when we share the notebook it can be re-executed. I will update it! Thanks for spotting this!